crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -98,22 +96,30 @@ pub(crate) fn use_def_map<'db>(db: &'db dyn Db, scope: ScopeId<'db>) -> Arc<UseD
 }
 
 /// Returns all attribute assignments for a specific class body scope.
-///
-/// Using [`attribute_assignments`] over [`semantic_index`] has the advantage that
-/// Salsa can avoid invalidating dependent queries if this scope's instance attributes
-/// are unchanged.
-#[salsa::tracked]
-pub(crate) fn attribute_assignments<'db>(
+pub(crate) fn attribute_assignments<'db, 's>(
```

Comment:
We should probably mention in the doc comment for this function that calling it adopts a direct dependency on the AST of the file containing `class_body_scope`, since we access the semantic index for that file.
```suggestion
///
/// Only call this when doing type inference on the same file as `class_body_scope`, otherwise it
/// introduces a direct dependency on that file's AST.
pub(crate) fn attribute_assignments<'db, 's>(
```